src/main/java/org/broadinstitute/hellbender/tools/spark/sv/GatherSplitReadsSpark.java

```diff
@@ -293,6 +412,130 @@ public GATKRead next()
         private static final int MIN_EVIDENCE = 5; // minimum evidence count in a cluster
     }
 
+    // Class that acts as a mapper from a stream of reads to a stream of <Kmer,BreakpointId> pairs.
+    private static final class ClusterKmerizer implements Iterator<Tuple2<Kmer,Long>>, Iterable<Tuple2<Kmer,Long>>
+    {
+        private static final int CLUSTER_BREAK_DISTANCE = 500;
+
+        public ClusterKmerizer( Integer partIdx, Iterator<GATKRead> someInputIterator, Set<Kmer> someKillSet )
+        {
+            inputIterator = someInputIterator;
+            killSet = someKillSet;
+            breakpointId = partIdx.longValue() << 32;
+        }
+
+        @Override
+        public boolean hasNext()
+        {
+            while ( !outputIterator.hasNext() )
+            {
+                if ( !inputIterator.hasNext() )
+                    return false;
+                outputIterator = processRead(inputIterator.next());
+            }
+            return true;
+        }
+
+        @Override
+        public Tuple2<Kmer,Long> next()
+        {
+            if ( !hasNext() )
+                throw new NoSuchElementException("ClusterKmerizer is exhausted.");
+            return outputIterator.next();
+        }
+
+        @Override
+        public Iterator<Tuple2<Kmer,Long>> iterator() { return this; }
+
+        private Iterator<Tuple2<Kmer,Long>> processRead( GATKRead read )
+        {
+            if ( read.getContig() != currentContig || read.getStart() > lastLocus+CLUSTER_BREAK_DISTANCE )
```

Comment:
And I'm back to understanding again, sorry about the back-and-forth. I'm a little worried that by setting a large cluster break distance, we'll end up with gaps in coverage in the reads that we pull out. Hopefully when we grab the mates of the suspicious reads we'll cover the gaps, I guess, but we might want to think about filling these regions in explicitly.